### DISCUSSION TITLE
Log Conext Displayed with Log Calls

### DISCUSSION THREAD

**Discussion by craftyshaun** [spatie/ray#257] [Ideas]
When Ray returns log calls it would be nice to get the additional context

For example when ray reacts to the following  in laravel `\Log::info('A Message','['context_1'=>true,'context_2'=>'A String']);` all that is retuned is  'A Message' along with the priority.

It would be nice if the content was dumped along with the message as the context usually holds important information about the error

From my look at the code, this will require changes to `\Spatie\Ray\Payloads\ApplicationLogPayload` to accept context as well as a message

**Comment by freekmurze** [comment]
Good suggestion. I'll take a look at this soon. If you want to speed things up, you could already submit a PR that adds support for this.

**Reply by craftyshaun** [reply]
Thanks, I had a quick look to try and work it out. But if you think it can be done by just editing the `laravel-ray` package or the `ray` package then I might try if I have time. I'll have to see how the Payloads work. Is there any doco or should I just explore other payloads?

**Comment by pieczarkazsolt** [comment]
hi @craftyshaun, are you working on it?

**Reply by craftyshaun** [reply]
Nope. I believe it needs changes to the desktop app because it hands back an `application_log` payload for the desktop app to render. 

The other way is to just make additional `ray()` calls looping over the context of the log event. I think that the change to ray desktop would be better then you could provide the key of the context entry and then the item with the correct renderer (ef JSON, Cabon).

The other option is to use the HTML payload and just render something ourself but I believe @freekmurze would want to keep the presentation in the desktop app rather than crafting it in the `ray` or `laravel-ray` package

@freekmurze may be able to provide some input.

**Reply by pieczarkazsolt** [reply]
I prefer the first solution instead of HTML payload. We send this:

```
         return [
            'value' => $this->message,
            'level' => $this->level,
            'context' => $this->context,
         ];
```
And the new values ​​must be displayed in the UI.

**Reply by craftyshaun** [reply]
It all depends on the render function of the app. It looks like most of the render logic is in the `ray` package.

Freek will have to provide guidance as the app is closed source.

**Reply by pieczarkazsolt** [reply]
Yes, i know it
